compiler/resolution/functionResolution.cpp

```diff
@@ -7885,10 +7885,16 @@ static void resolveExprMaybeIssueError(CallExpr* call) {
       if (foundDepthVal == false || arg->defPoint != fn->formals.tail) {
         var = toVarSymbol(paramMap.get(arg));
 
-        INT_ASSERT(var                        != NULL &&
+       if         (var                        != NULL &&
                    var->immediate             != NULL &&
-                   var->immediate->const_kind == CONST_KIND_STRING);
-
+                   var->immediate->const_kind != CONST_KIND_STRING) {
+        if (var->immediate->const_kind == NUM_KIND_BOOL) {
+          USR_FATAL_CONT(from, "bool arguments to compilerWarning()"
+            " and compilerError() must be cast to string");
+          str = astr(str, var->immediate->v_bool ? "true" : "false");
+        } else
+          INT_FATAL(call, "unexpected case");
```

Comment:
Why is bool treated specially here?  It seems like any argument other than a string should be cast to a string before being sent to these calls.